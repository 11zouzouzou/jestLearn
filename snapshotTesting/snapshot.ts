class Snapshot {
  a = { b: 11, c: "s", d: [11, "22"], e: { x: 11 } };
  constructor() {}
  toJSON() {
    return this.a;
  }
}

export { Snapshot };
