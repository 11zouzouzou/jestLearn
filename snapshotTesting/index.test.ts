import { Snapshot } from "./snapshot";

describe("snapshot test", () => {
  const snap = new Snapshot();
  test("one", () => {
    expect(snap.toJSON()).toMatchSnapshot();
  });
});
