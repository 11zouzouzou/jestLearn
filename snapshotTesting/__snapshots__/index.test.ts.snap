// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test one 1`] = `
Object {
  "b": 11,
  "c": "s",
  "d": Array [
    11,
    "22",
  ],
  "e": Object {
    "x": 11,
  },
}
`;
